Neos.Flow/Classes/Core/Booting/Scripts.php

```diff
@@ -356,10 +356,15 @@ public static function initializeCacheManagement(Bootstrap $bootstrap)
         $configurationManager = $bootstrap->getEarlyInstance(ConfigurationManager::class);
         $environment = $bootstrap->getEarlyInstance(Environment::class);
 
-        $cacheFactoryObjectConfiguration = $configurationManager->getConfiguration(ConfigurationManager::CONFIGURATION_TYPE_OBJECTS, CacheFactoryInterface::class);
-        $cacheFactoryClass = isset($cacheFactoryObjectConfiguration['className']) ? $cacheFactoryObjectConfiguration['className'] : CacheFactory::class;
+        $cacheFactoryClass = null;
+        $cacheFactoryObjectConfiguration = $configurationManager->getConfiguration(ConfigurationManager::CONFIGURATION_TYPE_OBJECTS);
+        foreach ($cacheFactoryObjectConfiguration as $objectConfiguration) {
+            if (isset($objectConfiguration[CacheFactoryInterface::class])) {
+                $cacheFactoryClass = $objectConfiguration[CacheFactoryInterface::class]['className'] ?? CacheFactory::class;
```

Comment:
```suggestion
        // workaround to find the correct cache manager implementation at compile time.
        // we can rely on the $objectConfiguration being ordered by the package names after its loading order.
        // normally this wiring would be done for proxy building a similar way, see ConfigurationBuilder.
        $cacheFactoryClass = CacheFactory::class;
        $cacheFactoryObjectConfiguration = $configurationManager->getConfiguration(ConfigurationManager::CONFIGURATION_TYPE_OBJECTS);
        foreach ($cacheFactoryObjectConfiguration as $objectConfiguration) {
            if (isset($objectConfiguration[CacheFactoryInterface::class]['className'])) {
                $cacheFactoryClass = $objectConfiguration[CacheFactoryInterface::class]['className'];
```

Reply:
With Marc Henrys comment, this is a bit more clear to me but I still find this change suspicious. We iterate over the all configurations and override the `$cacheFactoryClass` variable whenever a `className` is specified!?

> // we can rely on the $objectConfiguration being ordered by the package names after its loading order.

Is there really order guarantee? and is it correct that a _later_ value overrides a former (and if not, why don't we break the loop upon match)?

This screams for a test, but it's probably not so easy to test – given it happens so early in the bootstrap!?

Reply:
The configuration manager returns the configs in loading order, so a later appearing config overwrites the classname. Therefore I cannot break the loop early.
No clue right now on how to write a test for this change.

Reply:
Thanks for the clarification.
I might be able to help with the tests, but apart from those I feel that iterating over all object configurations for _every_ bootstrap is an overkill.. I hope that we can somehow access merged configuration or a find different way to configure the cache factory

Reply:
on the otherhand phps jit will probably do its magic. We loop over all flow packages that is not that much maybe 15?

Reply:
I'm more concerned about memory footprint than timing. For my (bare bones) Flow installation this is an array with 185 entries that is loaded into RAM.

Not sure if that's a problem.. But the whole thing makes me wonder: Why don't we merge object configurations like all the others?

Reply:
This currently happens in the `CompileTimeObjectManager` 😔

The array is anyway loaded as it is used later on by the ObjectManager?

Reply:
The array is structured by package i think to have a hint for debugging where what comes from

Reply:
Was that always the case? I mean why the hell do we separate the object configuration by package LOL. But that's for another PR / ticket

Reply:
Note: Yes was always the case, how unexpected 🤷 My bad. In that case I guess objects configuration is just bad for anything that needs to perform and we should switch to do this via settings. but also how the hell do we work with this in proxy building and such, why is it not merged aaahhhhhh

Reply:
the comment is wrong btw. !!! this muddies the waters we are searching for a cache FACTORY implementation as to potentially allow other implementations outside of neos/cache (our flow independent package). the Manager is "the manager", there should only be that specific one for flow. it could be final even in a future iteration.

Reply:
So with the error below, that's a nasty bit, we overwrite the factory implementation in Flow, I forgot about that part and the cacheManaegr is indeed not used in this overwritten factory, not sure why it exists in the first place... Bad decisions from the past come haunting LOL. I think what we do there in the constructor should just move to the manager and we should get rid of this implementation that implements stuff on top of the interface and thus breaks the static analysis below.